plugin/core/views.py

```diff
@@ -808,3 +808,52 @@ def format_completion(
     if item.get("textEdit"):
         completion.flags = sublime.COMPLETION_FLAG_KEEP_PREFIX
     return completion
+
+
+class CodeLensPhantom:
+    __slots__ = (
+        "region",
+        "view",
+        "lens",
+    )
+
+    def __init__(self, view: sublime.View, region: sublime.Region):
+        self.region = region
+        self.view = view
+        self.lens = []  #  type: List[str]
+
+    def __repr__(self) -> str:
+        return 'CodeLensPhantom(region={0.region!r}, lens={0.lens!r})'.format(self)
+
+    def add_annotation(self, annotation: str) -> None:
+        annotation = '<small style="font-family: system">{}</small>'.format(annotation)
+        self.lens.append(annotation)
+
+    def get_phantom_region(self) -> sublime.Region:
+        """Takes the current region and puts the phantom on the line above, properly indented.
+
+        Code-lenses are typically shown above the code in VSCode.
+        """
+        line = self.view.substr(self.view.line(self.region))
+        offset = 0
+        for ch in line:
+            if ch.isspace():
+                offset += 1
+            else:
+                break
+
+        row, col = self.view.rowcol_utf16(self.region.a)
+        point = self.view.text_point_utf16(max(row - 1, 0), offset, clamp_column=True)
+        return sublime.Region(point)
+
+    def to_phantom(self) -> sublime.Phantom:
+        if len(self.lens) == 1:
+            html = self.lens[0]
+        else:
+            html = '\n<small style="font-family: system">|</small>\n'.join(lens for lens in self.lens)
```

Comment:
Shouldn't this work?
```suggestion
            html = '\n<small style="font-family: system">|</small>\n'.join(self.lens)
```

Reply:
Yep. The original code wasn't a List[str] so I missed this during a refactoring.